### ISSUE TITLE
[Bug]: Freeze at World Creation

### ISSUE THREAD

**Issue by INooBTooMuch** [someaddons/dungeonbosses#42]
### Describe the bug you're experiencing

When I try to create a world the game freezes at the Loading X% screen and I can't exit minecaft.

### Reproducability

This is the [mods folder](https://gist.github.com/INooBTooMuch/10cfa142657308873bc94d017f336ec9) without Brutal Bosses - Dungeon Bosses installed and I am able to load into a world, might be a mod compatibility issue.

These are the Datapacks I am using :
Integrated Dungeons and Structures- Biomes O' Plenty Integration
Repurposed Structures - Better Dungeons Compat Datapack
Repurposed Structures - Domestication Innovation Compat Datapack
Repurposed Structures - Farmer's Delight Compat Datapack
Repurposed Structures - More Villagers Compat Datapack
And another one for Repurposed Structures but is not uploaded or something and it just stops the RS mineshafts from spawning.

### Mod up to date

- [X] Before submitting this issue I updated to the newest version and reproduced it

### Minecraft version

1.18

### Modloader version

Forge

### Logs and Datapacks

[Latest Log](https://gist.github.com/INooBTooMuch/720130964b01b5f616d05d9f62f3bbc8)

**Comment by Diversion98** [comment]
I got the same bug with no datapacks.
minecraft version 1.19.2 so different version then the author of this issue.

Disabling the mod fixes it.

any news on this issue?
need more logs?
I got 300 mods so compatibility might be an issue xD

**Comment by someaddons** [comment]
ye a log is always useful

**Comment by r8420** [comment]
I'm also having this issue with brutalbosses-1.18.2-5.6.jar on my server.
Currently using a lot of mods so it could be a compatibility issue. 
I attached the server logs.

[debug-3.log](https://github.com/someaddons/dungeonbosses/files/9997187/debug-3.log)
[latest.log](https://github.com/someaddons/dungeonbosses/files/9997189/latest.log)

**Comment by SiliconAshes** [comment]
Thank you for the time and effort you've put into this mod!

I am also having this issue - This is the 1.18.2 branch.  When I try to create a world the game freezes at the Loading X% screen.  The exact position seems to depend on the world seed. After some playing around, I found that my case seems to be an incompatibility with Cyclic.

Steps to reproduce:
1. Create a new 1.18.2 Minecraft instance
2. Install forge 40.1.86 (latest on MultMC)
3. Install brutalbosses-1.18.2-5.6 (latest on curseforge)
4. Install Cyclic-1.19.2-1.7.14 (latest on curseforge)
5. Create a new world.  I used the seed: AW-Lost

For me, the world gen progresses different amounts depending on world seed and then freezes.  AW-Lost progresses to ~84%.   The logs seem pretty nominal - right before the hang an my last attempt, the debug log has this, ten nothing after:

[11Dec2022 19:02:48.561] [Worker-Main-23/DEBUG] [com.brutalbosses.BrutalBosses/]: Spawning brutalbosses:pillagerboss at BlockPos{x=-553, y=-51, z=95} at Chest with:minecraft:chests/simple_dungeon in unkown structure.

As a side note.  It always fails if I add my who set of mods (~459) unless I take out Cyclic.  Then, it doesn't.  However, if I just have BrutalBosses and Cyclic, it does not fail every time - I suspect, as per the debug log message - only when trying to add a boss.

  I hope this info is useful.  I would have like to have a guaranteed failure case.  This isn't quite that.
